text/src/autogluon/text/automm/data/collator.py

```diff
@@ -0,0 +1,324 @@
+                "than the maximum length {} of the given arrays at axis={}.".format(
+                    max_length, max_arr_len, pad_axis
+                )
+            )
+        max_arr_len = max_length
+
+    size = arrs[0].size()
+    prev_trailing_dims = size[:pad_axis]
+    after_trailing_dims = size[pad_axis + 1 :]
+
+    out_dims = (len(arrs),) + prev_trailing_dims + (max_arr_len,) + after_trailing_dims
+    out_tensor = arrs[0].new_full(out_dims, pad_val)
+    for i, tensor in enumerate(arrs):
+        length = tensor.size(pad_axis)
+        out_tensor[i].narrow(pad_axis, 0, length)[:] = tensor
+
+    original_length = torch.as_tensor(original_length)
+
+    return out_tensor, original_length
+
+
+def _stack_arrs(arrs, use_shared_mem):
+    if isinstance(arrs[0], torch.Tensor):
+        if use_shared_mem:
+            elem = arrs[0]
+            numel = sum([x.numel() for x in arrs])
+            storage = elem.storage()._new_shared(numel)
+            out = elem.new(storage)
+            return torch.stack(arrs, 0, out=out)
+        else:
+            return torch.stack(arrs, 0)
+    else:
+        return _stack_arrs([torch.as_tensor(x) for x in arrs], use_shared_mem)
+
+
+class Stack:
+    r"""Stack the input data samples to construct the batch.
+
+    The N input samples must have the same shape/length and will be stacked to construct a batch.
+    """
+
+    def __call__(self, data):
+        """Collate the input data
+
+        Arguments:
+            data (list): The input data samples
+
+        Returns:
+            batch_data (torch.Tensor)
+        """
+        return _stack_arrs(data, True)
+
+
+class Pad:
+    r"""Returns a callable that pads and stacks data.
+
+    Arguments:
+        axis (int, optional): The axis to pad the arrays.
+            The arrays will be padded to the largest dimension at :attr:`axis`.
+            For example, assume the input arrays have shape (10, 8, 5), (6, 8, 5), (3, 8, 5)
+            and the `axis` is 0.
+            Each input will be padded into (10, 8, 5) and then stacked to form the final output,
+            which has shape（3, 10, 8, 5). (default ``0``)
+        pad_val (float or int, optional): The padding value. (default ``0``)
+        round_to (int, optional):
+            If specified, the padded dimension will be rounded to be multiple of this argument.
+            Mutually exclusive with :attr:`max_length`. (default ``None``)
+        max_length (int, optional):
+            If specified, the padded dimension will have length :attr:`max_length`,
+            and it must be larger than the maximum length in the arrays at :attr:`axis`.
+            Mutually exclusive with :attr:`round_to`.  (default ``None``)
+        ret_length (bool, optional): Whether to return the valid length in the output.
+            (default ``False``)
+    """
+
+    def __init__(self, axis=0, pad_val=None, round_to=None, max_length=None, ret_length=False):
```

Comment:
nit: `pad_val=0` default

Reply:
None will be handled later by `self._pad_val = 0 if pad_val is None else pad_val`. But following your suggestion, I have changed the default to 0 since it makes more sense.